### PR TITLE
Python: Add store and metadata properties to OpenAIPromptExecutionSettings

### LINKED ISSUE #9918: .Net: Add `store` and `metadata` properties to `OpenAIPromptExecutionSettings` in OpenAI Connector
The recently added [store](https://platform.openai.com/docs/api-reference/chat/create#chat-create-store) and [metadata](https://platform.openai.com/docs/api-reference/chat/create#chat-create-metadata) chat completion options are both used for [model distillation](https://platform.openai.com/docs/guides/distillation) and [evals](https://platform.openai.com/docs/guides/evals), and it would be useful and convenient if they could be added as properties to `OpenAIPromptExecutionSettings`.

Both properties have been added to the official OpenAI .net SDK [here](https://github.com/openai/openai-dotnet/blob/c046c34ad06f7ca050b2f8044cfc8372d8cf0e76/src/Custom/Chat/ChatCompletionOptions.cs#L163C5-L178C51)

### PR DESCRIPTION
**Pull request by markwallace-microsoft**
### Motivation and Context

Closes #9918 

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

**Review by eavanvalkenburg — APPROVED** [review]

File: python/samples/concepts/chat_completion/simple_chatbot_store_metadata.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+import asyncio
+
+from samples.concepts.setup.chat_completion_services import (
+    Services,
+    get_chat_completion_service_and_request_settings,
+)
+from semantic_kernel.contents import ChatHistory
+
+# This sample shows how to create a chatbot that whose output can be stored for use with the OpenAI
```
Comment (TaoChenOSU) [inline python/samples/concepts/chat_completion/simple_chatbot_store_metadata.py:11]:
```suggestion
# This sample shows how to create a chatbot whose output can be stored for use with the OpenAI
```

File: python/samples/concepts/chat_completion/simple_chatbot_store_metadata.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+# model distillation or evals products.
+# This sample uses the following three main components:
```
Comment (TaoChenOSU) [inline python/samples/concepts/chat_completion/simple_chatbot_store_metadata.py:13]:
```suggestion
# This sample uses the following two main components:
```

File: python/samples/concepts/chat_completion/simple_chatbot_store_metadata.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+# This is the system message that gives the chatbot its personality.
+system_message = """
+You are a chat bot whose expertise is basketball.
+Your name is Mosscap and you have one goal: to answer questions about basketball.
+"""
+
+# Create a chat history object with the system message.
+chat_history = ChatHistory(system_message=system_message)
+# Configure the store amd metadata settings for the chat completion service.
```
Comment (moonbox3) [inline python/samples/concepts/chat_completion/simple_chatbot_store_metadata.py:35]: A small typo: "amd" -> "and"?

Comment (markwallace-microsoft) [reply python/samples/concepts/chat_completion/simple_chatbot_store_metadata.py:35]: Will fix in my next PR

**Pull request merged**